google_photos_takeout_helper/__main__.py

```diff
@@ -255,15 +248,109 @@ def set_file_exif_date(file, creation_date):
         try:
             _piexif.insert(_piexif.dump(exif_dict), file)
         except Exception as e:
-            print('Couldnt insert exif!')
+            print("Couldn't insert exif!")
             print(e)
 
-
     def get_date_str_from_json(json):
         return _datetime.fromtimestamp(
             int(json['photoTakenTime']['timestamp'])
         ).strftime('%Y:%m:%d %H:%M:%S')
 
+    def change_to_rational(number):
+        """convert a number to rantional
+        Keyword arguments: number
+        return: tuple like (1, 2), (numerator, denominator)
+        """
+        f = Fraction(str(number))
+        return f.numerator, f.denominator
+
+    # got this here https://github.com/hMatoba/piexifjs/issues/1#issuecomment-260176317
+    def degToDmsRational(degFloat):
+        min_float = degFloat % 1 * 60
+        sec_float = min_float % 1 * 60
+        deg = math.floor(degFloat)
+        deg_min = math.floor(min_float)
+        sec = round(sec_float * 100)
+
+        return [(deg, 1), (deg_min, 1), (sec, 100)]
+
+    def set_file_geo_data(file, json):
+        """
+        Reads the geoData from google and saves it to the EXIF. This works assuming that the geodata looks like -100.12093, 50.213143. Something like that.
+
+        Written by DalenW.
+        :param file:
+        :param json:
+        :return:
+        """
+
+        # prevents crashes
+        try:
+            exif_dict = _piexif.load(file)
+        except (_piexif.InvalidImageDataError, ValueError):
+            exif_dict = {'0th': {}, 'Exif': {}}
+
+        # fetches geo data from the photos editor first.
+        longitude = float(json['geoData']['longitude'])
+        latitude = float(json['geoData']['latitude'])
+        altitude = float(json['geoData']['altitude'])
+
+        # fallbacks to GeoData Exif if it wasn't set in the photos editor.
+        # https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/pull/5#discussion_r531792314
+        if longitude == 0:
+            longitude = float(json['geoDataExif']['longitude'])
+        if latitude == 0:
+            latitude = float(json['geoDataExif']['latitude'])
+        if altitude == 0:
+            altitude = float(json['geoDataExif']['altitude'])
```

Comment:
We just wanted to check if the `geoData` was blank, not check if every value was 0:
```suggestion
        longitude = float(json['geoData']['longitude'])
        latitude = float(json['geoData']['latitude'])
        altitude = json['geoData']['altitude']
        # Prioritise geoData set from GPhotos editor
        if longitude == 0 and latitude == 0:
            longitude = float(json['geoDataExif']['longitude'])
            latitude = float(json['geoDataExif']['latitude'])
            altitude = json['geoDataExif']['altitude']
```

Reply:
I mean, your suggested changes do pretty much the same thing with one less if check. The reason why I added the fallback is because I'm not sure if what an edited location history looks like. Does it copy over the altitude if you change the latitude? What if just the latitude is changed, then does it change the longitude? Unlikely, but better safe than sorry.

Reply:
From what I've seen, `geoData` is usually identical to `geoDataExif`, and it's different only if user edited it in GPhotos editor

That's why I wanted to chose only one or another :+1: